### PR TITLE
[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70475
* #70375
* #70374

Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

File: torch/csrc/jit/mobile/nnc/aot_compiler.cpp [hunk 1]
```diff
@@ -313,9 +314,26 @@ c10::IValue preprocess(
...
+      LLVMTargetAttrs() = "";
+    } else if (arch == "aarch64") {
+      LLVMTargetTriple() = "aarch64-unknown-unknown";
+      LLVMTargetAttrs() = "+neon,+thumb2";
+    } else if (arch == "arm") {
+      LLVMTargetTriple() = "arm-unknown-unknown";
+      LLVMTargetAttrs() = "+neon,+thumb2";
+    } else if (arch == "") {
+      // Do nothing
+    } else {
```
Comment (priyaramani) [inline torch/csrc/jit/mobile/nnc/aot_compiler.cpp:334]: Can we add x86 (aka x86-32) to the list too?

File: torch/csrc/jit/mobile/nnc/aot_compiler.cpp [hunk 2]
```diff
@@ -313,9 +314,26 @@ c10::IValue preprocess(
...
     std::cout << "Asm file name: " << asmfile_name << "\n";
+    std::cout << "Arch: " << arch << "\n";
+
+    if (arch == "x86-64") {
+      LLVMTargetTriple() = "x86_64-unknown-unknown";
+      LLVMTargetAttrs() = "";
+    } else if (arch == "aarch64") {
+      LLVMTargetTriple() = "aarch64-unknown-unknown";
+      LLVMTargetAttrs() = "+neon,+thumb2";
+    } else if (arch == "arm") {
```
Comment (priyaramani) [inline torch/csrc/jit/mobile/nnc/aot_compiler.cpp:329]: I guess this is for all armv7, armv8 etc, right?

**Review by priyaramani — APPROVED** [review]

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

**Pull request closed without merging**